### PR TITLE
Implements Mute idiom

### PR DESCRIPTION
**Pull request by npathai**
Implements #385

Provided two flavors of mute
 - `mute()` an `AssertionError` on exception. Example of `ByteArrayOutputStream` bulk write.
 - `loggedMute()`  suppression of exception with a log. Example of closing a connection. (This is outdated in Java 7 where we can use try with resources. But applies to similar operations such as cleaning up resources.)

### CODE REVIEW

**Comment by iluwatar** [comment]
Review comments:
- `README.md` is missing

File: mute-idiom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	</parent>
+	<artifactId>mute-idiom</artifactId>
+	<dependencies>
+		<dependency>
+			<groupId>junit</groupId>
+			<artifactId>junit</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+        <groupId>org.mockito</groupId>
```
Comment (iluwatar) [inline mute-idiom/pom.xml]: Check indentation

**Comment by iluwatar** [comment]
@npathai you have my review comments.

**Comment by npathai** [comment]
@iluwatar Done with the changes.

**Comment by iluwatar** [comment]
@npathai still can't see `README.md`. The other changes are good.

**Comment by npathai** [comment]
@iluwatar Pardon me, I forgot to add it in my commit. Its done now.

**Comment by iluwatar** [comment]
It's a good trick to have up in your sleeve :) Well done example @npathai

**Pull request merged**